src/dune_rules/coq/coq_rules.ml

```diff
@@ -386,21 +391,37 @@ let setup_coqc_rule ~loc ~dir ~sctx ~coqc_dir ~file_targets ~stanza_flags
     coq_module =
   (* Process coqdep and generate rules *)
   let* boot_type = boot_type ~dir ~use_stdlib ~wrapper_name coq_module in
-  let deps_of = deps_of ~dir ~use_stdlib ~wrapper_name coq_module in
+  (* let deps_of = deps_of ~dir ~use_stdlib ~wrapper_name coq_module in *)
   let* coqc = coqc ~loc ~dir ~sctx in
-  let target_obj_files =
-    Command.Args.Hidden_targets
-      (Coq_module.obj_files ~wrapper_name ~mode ~obj_files_mode:Coq_module.Build
-         ~obj_dir:dir coq_module
-      |> List.map ~f:fst)
+  let obj_files =
+    Coq_module.obj_files ~wrapper_name ~mode ~obj_files_mode:Coq_module.Build
+      ~obj_dir:dir coq_module
+    |> List.map ~f:fst
   in
+  let target_obj_files = Command.Args.Hidden_targets obj_files in
   let* args =
     generic_coq_args ~sctx ~dir ~wrapper_name ~boot_type ~stanza_flags ~ml_flags
       ~theories_deps ~theory_dirs ~mode ~coq_prog:`Coqc coq_module
   in
+  let open Action_builder.O in
+  let vo_target =
+    Path.Build.set_extension ~ext:".vo" (Coq_module.source coq_module)
+  in
```

Comment:
This logic with .vo extension should be hidden in the `Dep_map` module as we do in Coq's own rules.

Reply:
Hidden in new PR